cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -493,6 +499,23 @@ TYPED_TEST(TestStringKernels, Utf8Lower) {
                                   CallFunction("utf8_lower", {invalid_input}));
 }
 
+TYPED_TEST(TestStringKernels, Utf8SwapCase) {
+  this->CheckUnary("utf8_swapcase", "[\"aAazZæÆ&\", null, \"\", \"b\"]", this->type(),
+                   "[\"AaAZzÆæ&\", null, \"\", \"B\"]");
```

Comment:
Same here: add characters that don't have a casing.

Reply:
Also addressed in [8f65c70](https://github.com/apache/arrow/pull/10855/commits/8f65c700ae86d03d594dcf8957f7f17a583b2a97)